onnxruntime/core/framework/utils.cc

```diff
@@ -795,7 +795,7 @@ common::Status ExecuteGraph(const SessionState& session_state,
                                   logger);
   }
 #endif
-  bool synchronize_execution_providers = std::stoi(run_options.config_options.GetConfigOrDefault(kOrtRunOptionsConfigSynchronizeExecutionProviders, "1"));
+  bool synchronize_execution_providers = run_options.config_options.GetConfigOrDefault(kOrtRunOptionsConfigDisableSynchronizeExecutionProviders, "").empty();
```

Comment:
what if user set to '0'?

Reply:
I had the same question. But the same can be said about `false` and `true` as string. Ideally there would be one string to bool function somewhere for reuse or would you prefer the `stoi` cast that i had before ?

Reply:
I'll change to `== "0"` as this seems to be the often used case. https://github.com/microsoft/onnxruntime/blob/7a6e5cfccd520dd5b17abd537d4e394384fcd1d8/onnxruntime/core/session/inference_session.cc#L258